crypto/fipsmodule/bn/rsaz_exp_x2.c

```diff
@@ -346,47 +268,49 @@ int RSAZ_mod_exp_x2_ifma256(BN_ULONG *out,
  */
 # define DAMS(r,a,m,k0) damm((r),(a),(a),(m),(k0))
 
-    switch (modulus_bitsize) {
+    switch (modlen) {
     case 1024:
         red_digits = 20;
         exp_digits = 16;
-        damm = ossl_rsaz_amm52x20_x2_ifma256;
-        extract = ossl_extract_multiplier_2x20_win5;
+        damm = rsaz_amm52x20_x2_ifma256;
+        extract = extract_multiplier_2x20_win5;
         break;
     case 1536:
         // Extended with 2 digits padding to avoid mask ops in high YMM register 
         red_digits = 30 + 2;
         exp_digits = 24;
-        damm = ossl_rsaz_amm52x30_x2_ifma256;
-        extract = ossl_extract_multiplier_2x30_win5;
+        damm = rsaz_amm52x30_x2_ifma256;
+        extract = extract_multiplier_2x30_win5;
         break;
     case 2048:
         red_digits = 40;
         exp_digits = 32;
-        damm = ossl_rsaz_amm52x40_x2_ifma256;
-        extract = ossl_extract_multiplier_2x40_win5;
+        damm = rsaz_amm52x40_x2_ifma256;
+        extract = extract_multiplier_2x40_win5;
         break;
     default:
         goto err;
     }
 
-    storage_len_bytes = (2 * red_digits                         /* red_Y     */
-                       + 2 * red_digits                         /* red_X     */
-                       + 2 * red_digits * (1U << exp_win_size)  /* red_table */
-                       + 2 * (exp_digits + 1))                  /* expz      */
-                       * sizeof(BN_ULONG)
-                       + 64;                                    /* alignment */
+    // allocate space for 2x num digits, aligned because the data in
+    // the vectors need to be 64-bit aligned.
+    storage_len_bytes = (2 * red_digits                    /* red_Y     */
+                       + 2 * red_digits                    /* red_X     */
+                       + 2 * red_digits * exp_win_size_bit /* red_table */
+                       + 2 * (exp_digits + 1))             /* expz      */
```

Comment:
My understanding here is that `exp_digits` is 0-_padded_. The 0's are written offset by 1. I believe this is because when the exponent is chunked into indices for the table, and the total radix of the exponent is not divisible evenly by chunk size ($w$ the "w-ary exponentiation" algorithm cited here), the padding allows for including leading zeros in the most significant chunk because the algorithm likes to treat each chunk as equal-width, i.e. $w$.